SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1233,16 +1261,24 @@ int32 USpatialNetDriver::ServerReplicateActors_PrioritizeActors(UNetConnection*
 			DeletedCount++;
 		}
 
-		// Sort by priority
-		Sort(OutPriorityActors, FinalSortedCount, FCompareFActorPriority());
+		if(MigrationHandler.GetActorsToMigrate().Num() > 0)
+		{
+			// Process actors migrating first, in order to not have them separated if they need to migrate together, and replication rate limiting happens.
+			Sort(OutPriorityActors, FinalSortedCount, FCompareFActorPriorityAndMigration(MigrationHandler));
```

Comment:
Why does it matter if they're separated in a list that executes sequentially?

Reply:
I want to put it at the front, in case rate limiting stops replication halfway. I do not want some actors split from the other from the same hierarchy.